includes/plugins/woocommerce-subscriptions/class-on-hold-duration.php

```diff
@@ -119,27 +120,37 @@ public static function maybe_apply_on_hold_duration_rule( $retry_rules ) {
 	}
 
 	/**
-	 * Conditionally schedule expiration action.
+	 * Conditionally schedule expiration action if subscription is manual and/or retry system is disabled.
 	 *
-	 * @param int    $order_id     Order ID.
 	 * @param object $subscription Subscription object.
 	 */
-	public static function maybe_schedule_expiration( $order_id, $subscription ) {
-		if ( 'on-hold' === $subscription->get_status() ) {
-			$default_grace_period = 7 * DAY_IN_SECONDS; // 7 days, the number of days the retry system normally waits before marking a subscription as expired.
+	public static function maybe_schedule_expiration( $subscription ) {
+		if ( $subscription->is_manual() || ! \WCS_Retry_Manager::is_retry_enabled() ) {
```

Comment:
Note to self: this should also triggered for non-manual subscriptions.

Reply:
Fixed in https://github.com/Automattic/newspack-plugin/pull/3710/commits/c900f8b6b1c60caf1fd259fa486b94e2b0588fd3